src/server/scripts/EasternKingdoms/ZulGurub/boss_mandokir.cpp

```diff
@@ -251,31 +303,70 @@ class boss_mandokir : public CreatureScript
                             Talk(SAY_OHGAN_DEAD);
                         }
                         else
+                        {
                             events.ScheduleEvent(EVENT_CHECK_OHGAN, 1000);
+                        }
                         break;
                     case EVENT_WATCH_PLAYER:
                         if (Unit* player = SelectTarget(SelectTargetMethod::Random, 0, 100, true))
                         {
                             DoCast(player, SPELL_WATCH);
                             Talk(SAY_WATCH, player);
                         }
-                        events.ScheduleEvent(EVENT_WATCH_PLAYER, urand(12000, 15000));
+                        events.ScheduleEvent(EVENT_WATCH_PLAYER, urand(12000, 24000));
                         break;
                     case EVENT_CHARGE_PLAYER:
                         DoCast(SelectTarget(SelectTargetMethod::Random, 0, 40, true), SPELL_CHARGE);
-                        events.ScheduleEvent(EVENT_CHARGE_PLAYER, urand(22000, 30000));
+                        events.ScheduleEvent(EVENT_FRIGHTENING_SHOUT, 1500);
+                        if (Unit* mainTarget = SelectTarget(SelectTargetMethod::MaxThreat, 0, 100.0f))
+                        {
+                            me->GetThreatMgr().modifyThreatPercent(mainTarget, -100);
+                        }
+                        events.ScheduleEvent(EVENT_CHARGE_PLAYER, urand(30000, 40000));
                         break;
+                    case EVENT_EXECUTE:
+                        if (me->GetVictim() && me->GetVictim()->HealthBelowPct(20))
+                        {
+                            DoCastVictim(SPELL_EXECUTE, true);
+                        }
+                        events.ScheduleEvent(EVENT_EXECUTE, urand(7000, 14000));
+                        break;
+                    case EVENT_FRIGHTENING_SHOUT:
+                        DoCastAOE(SPELL_FRIGHTENING_SHOUT);
+                        break;
+                    case EVENT_CLEAVE:
+                        {
+                            std::list<Unit*> meleeRangeTargets;
+                            auto i = me->GetThreatMgr().getThreatList().begin();
+                            for (; i != me->GetThreatMgr().getThreatList().end(); ++i)
+                            {
+                                Unit* target = (*i)->getTarget();
+                                if (me->IsWithinMeleeRange(target))
+                                {
+                                    meleeRangeTargets.push_back(target);
+                                }
+                            }
+                            if (meleeRangeTargets.size() >= 5)
+                            {
+                                if (Unit* currentTarget = me->GetVictim())
+                                {
+                                    me->CastSpell(currentTarget, SPELL_MANDOKIR_CLEAVE, false);
+                                }
```

Comment:
```suggestion
                                if (Unit* currentTarget = me->GetVictim())
                                {
                                    DoCastVictim(SPELL_MANDOKIR_CLEAVE);
                                }
```

Reply:
read

Reply:
![image](https://user-images.githubusercontent.com/47818697/171657852-131a3fe4-485b-44f5-a683-a4aa183daf1b.png)
![image](https://user-images.githubusercontent.com/47818697/171657987-c13f66f3-371e-47e4-9a4d-216383a585b6.png)

It's exactly the same thing. How about you read it yourself?